AlgebraInLean/World01/Sheet4.lean

```diff
@@ -0,0 +1,155 @@
+import AlgebraInLean.World01.Sheet3
+
+namespace AlgebraInLean
+
+/- Now that we know how to define a group, and some basic properties of groups,
+we can find important examples of groups-/
+
+/- The group you are likely most familiar with is the integers, with addition
+as the group operation.-/
+instance : Group ℤ where
+  op := fun x y => x + y
+
+  op_assoc := by -- Mathlib already has a proof that the addition is associative, try to use that
+    -- sorry
+    -- SAMPLE SOLUTION
+    intro a b c
+    exact Int.add_assoc a b c
+    -- END OF SAMPLE SOLUTION
+
+  id :=
+    -- sorry
+    -- SAMPLE SOLUTION
+    0
+    -- END OF SAMPLE SOLUTION
+
+  op_id := by
+    -- sorry
+    -- SAMPLE SOLUTION
+    intro a
+    exact Int.add_zero a
+    -- END OF SAMPLE SOLUTION
+
+  id_op := by
+    -- sorry
+    -- SAMPLE SOLUTION
+    intro a
+    exact Int.zero_add a
+    -- END OF SAMPLE SOLUTION
+
+  inv :=
+    -- sorry
+    -- SAMPLE SOLUTION
+    fun a => -a
+    -- END OF SAMPLE SOLUTION
+
+  inv_op := by
+    -- sorry
+    -- SAMPLE SOLUTION
+    intro a
+    exact Int.add_left_neg a
+    -- END OF SAMPLE SOLUTION
+
+/- Structures other than numbers also have the group properties. Take the
+symmetries of rotating a polygon, for example. Each element could be a
+rotation that maintains the symmetries of the polygon, and the group operation
+can be composition of these rotations. For example, consider the rotations of
+a triangle. All symmetric rotations must be multiples of 120°, of which there
+are three: {0°, 120°, 240°}. Any other rotations can be written as one of
+these three and some amount of 360° rotations. Now, we can prove that
+this set, commonly called C3, is a group.-/
```

Comment:
C3 isn't the group of symmetries of a triangle. It's the group of rotations of a triangle. Symmetries contains reflections too (D3 instead of C3)